base/abstractarray.jl

```diff
@@ -1198,7 +1198,8 @@ function copymutable(a::AbstractArray)
 end
 copymutable(itr) = collect(itr)
 
-zero(x::AbstractArray{T}) where {T} = fill!(similar(x, typeof(zero(T))), zero(T))
+zero(x::AbstractArray{T}) where {T<:Number} = fill!(similar(x, typeof(zero(T))), zero(T))
+zero(x::AbstractArray) = map(zero, x)
```

Comment:
Should this use broadcasting instead, as custom sparse arrays might define their own broadcast styles and don't need to iterate over the entire array?

Reply:
maybe? idk

Custom sparse arrays should probably be implementing their own `zero` function outright.
Since they likely have their own opinions about whether or not to store structural zeros etc (the SparseArrays stdlib does this).

Reply:
It's also a whole lot easier to define a single-argument `map(f, ::CustomArray)` than it is to define your own broadcast style. I think the map is fine; it's straightforward and obvious and even enables the laser-focused `map(::typeof(zero), ::CustomArray)`.